src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Write.Helpers.cs

```diff
@@ -54,7 +54,7 @@ private static void VerifyValueAndType(object value, Type type)
             }
         }
 
-        private static byte[] WriteCoreBytes(object value, Type type, JsonSerializerOptions options)
+        private static byte[] WriteCoreBytes(object? value, Type? type, JsonSerializerOptions? options)
```

Comment:
Similarly here and else where (WriteCoreString/WriteValueCore/WriteCore methods). Type shouldn't be nullable.